File: logistics_management/logistics_management/report/job_detail_report/job_detail_report.js
// Copyright (c) 2024, siva and contributors
// For license information, please see license.txt

frappe.query_reports["Job Detail Report"] = {
    "filters": [
        {
            "fieldname": "customer",
            "label": __("Customer"),
            "fieldtype": "Link",
            "options": "Customer"
        },
        {
            "fieldname": "job_status",
            "label": __("Job Status"),
            "fieldtype": "Select",
            "options": "\nPending\nIn Progress\nCompleted\nOn Hold\nCancelled"
        },
        {
            "fieldname": "created_user",
            "label": __("Created User"),
            "fieldtype": "Link",
            "options": "User"
        },
        {
            "fieldname": "sales_person",
            "label": __("Sales Person"),
            "fieldtype": "Link",
            "options": "Sales Partner"
        },
        // {
        //     "fieldname": "company",
        //     "label": __("Company"),
        //     "fieldtype": "Link",
        //     "options": "Company",
        //     "default": frappe.defaults.get_user_default("company")
        // }
    ]
};

File: logistics_management/logistics_management/report/job_detail_report/job_detail_report.py
# Assuming this code is placed in a file named job_detail_report.py within your custom app
import frappe
from frappe import _, msgprint

def get_columns():
    return [
        {
            'fieldname': 'name',
            'label': _('Name'),
            'fieldtype': 'Link',
            'options': 'Job Details',
            'width': '200'
        },
        {
            'fieldname': 'customer',
            'label': _('Customer'),
            'fieldtype': 'Link',
            'options': 'Customer',
            'width': '200'
        },
        
        {
            'fieldname': 'job_status',
            'label': _('Job Status'),
            'fieldtype': 'Select',
            'options': 'Pending\nIn Progress\nCompleted\nOn Hold\nCancelled',
            'width': '250'
        },
        {
            'fieldname': 'created_user',
            'label': _('Created User'),
            'fieldtype': 'Link',
            'options': 'User',
            'width': '250'
        },
        {
            'fieldname': 'sales_person',
            'label': _('Sales Person'),
            'fieldtype': 'Link',
            'options': 'Sales Partner',
            'width': '200'
        },
        {
            'fieldname': 'amount',
            'label': _('Amount'),
            'fieldtype': 'Float',
            'width': '100'


        }
    ]

def get_data(filters):
    fields = ['name','job_id', 'job_status', 'created_user', 'sales_person','amount']

    data = frappe.get_all('Job Details', fields=fields, filters=filters)

    return data


def execute(filters=None):
    if not filters:
        filters = {}

    columns = get_columns()
    data = get_data(filters)

    return columns, data
